src/dma.rs

```diff
@@ -480,27 +495,30 @@ pub trait Transmit {
     type ReceivedWord;
 }
 
+/// Trait for circular DMA readings from peripheral to memory.
 pub trait CircReadDma<B, RS>: Receive
 where
-    B: as_slice::AsMutSlice<Element = RS>,
+    &'static mut [B; 2]: StaticWriteBuffer<Word = RS>,
+    B: 'static,
     Self: core::marker::Sized,
 {
     fn circ_read(self, buffer: &'static mut [B; 2]) -> CircBuffer<B, Self>;
 }
 
+/// Trait for DMA readings from peripheral to memory.
 pub trait ReadDma<B, RS>: Receive
 where
-    B: as_slice::AsMutSlice<Element = RS>,
-    Self: core::marker::Sized,
+    B: StaticWriteBuffer<Word = RS>,
+    Self: core::marker::Sized + TransferPayload,
```

Comment:
Seeing this code, I feel like `TransferPayload` is a missleading name for this trait. Something like "DmaPeripheral" would make more sense to me.

Though we can deal with that later I suppose

Reply:
Today I think we only put peripherals in there, yes. However, the generic name is good in case someone needs to put more stuff in there, but yeah, maybe we can get a better name than `..Payload`.

Reply:
Right, the genericity is a good point. I can't really think of a good name then :/

Maybe something like TransferSource, though that doesn't work when it's the target of the transfer